fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/FabricSkyPropertyBuilder.java

```diff
@@ -0,0 +1,240 @@
+		return this;
+	}
+
+	/**
+	 * Should the sky be slightly brighter.
+	 *
+	 * @param alternateSkyColor should sky be brighter
+	 * @return a reference to the FabricItemGroupBuilder
+	 */
+	public FabricSkyPropertyBuilder alternateSkyColor(boolean alternateSkyColor) {
+		this.alternateSkyColor = alternateSkyColor;
+		return this;
+	}
+
+	/**
+	 * Transforms the given fog color based on the current height of the sun. This is used in vanilla to darken fog during night.
+	 *
+	 * @param adjustFogColor Spits out an adjusted RBGA value based on previous color and the sun height.
+	 * @return a reference to the FabricItemGroupBuilder
+	 */
+	public FabricSkyPropertyBuilder adjustFogColor(AdjustFogColor adjustFogColor) {
+		this.adjustFogColor = adjustFogColor;
+		return this;
+	}
+
+	/**
+	 * Tells the client if thick fog should be render at xy coordinates.
+	 *
+	 * @param useThickFog Used to etermines if fog should be thick at xy camera coordinate
+	 * @return a reference to the FabricItemGroupBuilder
+	 */
+	public FabricSkyPropertyBuilder useThickFog(UseThickFog useThickFog) {
+		Objects.requireNonNull(useThickFog);
+		this.useThickFog = useThickFog;
+		return this;
+	}
+
+	/**
+	 * This is used to tell Vanilla what kind of sky to render.
+	 *
+	 * @param skyType Type of vanilla sky to render
+	 * @return a reference to the FabricItemGroupBuilder
+	 */
+	public FabricSkyPropertyBuilder skyType(SkyProperties.SkyType skyType) {
+		Objects.requireNonNull(skyType);
+		this.skyType = skyType;
+		return this;
+	}
+
+	/**
+	 * Will cause lighting to be higher.
+	 *
+	 * @param brightenLighting should lighting be higher.
+	 * @return a reference to the FabricItemGroupBuilder
+	 */
+	public FabricSkyPropertyBuilder brightenLighting(boolean brightenLighting) {
+		this.brightenLighting = brightenLighting;
+		return this;
+	}
+
+	/**
+	 * This is used in vanilla to render sunset and sunrise fog.
+	 * Function input is an array for storing color, current sky angle, and tick delta
+	 * @param fogColorOverride function used to calculate fog color
+	 * @return a reference to the FabricItemGroupBuilder
+	 */
+	public FabricSkyPropertyBuilder fogColorOverride(FogColorOverride fogColorOverride) {
+		Objects.requireNonNull(fogColorOverride);
+		this.fogColorOverride = fogColorOverride;
+		return this;
+	}
+
+	/**
+	 * Says if lighting should be darker.
+	 *
+	 * @param darkened should lighting be darker
+	 * @return a reference to the FabricItemGroupBuilder
+	 */
+	public FabricSkyPropertyBuilder darkened(boolean darkened) {
+		this.darkened = darkened;
+		return this;
+	}
+
+	/**
+	 * Create an instance of the SkyProperties.
+	 *
+	 * @return An instance of the built SkyProperties
+	 */
+	public SkyProperties build() {
+		return new FabricSkyproperties(cloudsHeight, alternateSkyColor, skyType, brightenLighting, darkened, fogColorOverride, adjustFogColor, useThickFog);
+	}
+
+	public static class FabricSkyproperties extends SkyProperties {
```

Comment:
Is this class supposed to be visible to users? If not making it private or package private is better, so we can change it freely like it's an implementation detail.

Reply:
Its suppose to implementation.

Reply:
Then I suggest
```suggestion
	private static class FabricSkyproperties extends SkyProperties {
```
so modders cannot use it in instanceof checks, call its methods etc. explicitly, and we can change this class as part of the implementaion however we want to.

Reply:
Done.